src/seedu/addressbook/logic/Logic.java

```diff
@@ -71,7 +71,10 @@ protected void setLastShownList(List<? extends ReadOnlyPerson> newList) {
     public CommandResult execute(String userCommandText) throws Exception {
         Command command = new Parser().parseCommand(userCommandText);
         CommandResult result = execute(command);
-        recordResult(result);
+        storage.save(addressBook);
+        if(command.isMutating() == true) {
```

Comment:
There's no need to compare to `true`, `if(command.isMutating())` should be fine. 😄